Day29/LinhTTK/student-listing/src/Components/Layout/Page.js

```diff
@@ -0,0 +1,268 @@
+  const [show, setShow] = useState(5);
+  const [name, setName] = useState('');
+  const [sex, setGender] = useState('gender');
+  const [age, setAge] = useState('');
+  const [search, setSearch] = useState(false);
+  useEffect(() => {
+    getStudent()
+      .then((e) => e.data)
+      .then((data) => setStudent(data))
+      .catch((err) => err);
+  }, []);
+  if (localStorage.getItem('user') === '') {
+    alert('Chưa đăng nhập');
+    history.push('/Login');
+  }
+  const handleClick = () => {
+    setShow(0);
+    setSearch(!search);
+  };
+  return (
+    <>
+      <main>
+        <Header />
+        <section className=" container mt-5">
+          <div className=" d-flex justify-content-end">
+            <InputGroup
+              className="mb-3 mx-3"
+              onChange={(e) => setName(e.target.value.trim().toLowerCase())}
+              value={name}
+            >
+              <FormControl
+                placeholder="Search name"
+                aria-label="Search name"
+                aria-describedby="basic-addon1"
+              />
+            </InputGroup>
+            <Dropdown>
+              <Dropdown.Toggle variant="variant" className=" border mx-3">
+                {sex}
+              </Dropdown.Toggle>
+              <Dropdown.Menu>
+                <Dropdown.Item onClick={() => setGender('Male')}>
+                  Male
+                </Dropdown.Item>
+                <Dropdown.Item onClick={() => setGender('Female')}>
+                  Female
+                </Dropdown.Item>
+              </Dropdown.Menu>
+            </Dropdown>
+            <InputGroup
+              className="mb-3 mx-3"
+              onChange={(e) => {
+                setAge(e.target.value.trim());
+              }}
+              value={age}
+            >
+              <FormControl
+                placeholder="Age"
+                aria-label="Age"
+                aria-describedby="basic-addon1"
+              />
+            </InputGroup>
+            {search ? (
+              <Button
+                variant="outline-dark"
+                className=" active"
+                onClick={() => handleClick()}
+              >
+                <i className="fas fa-search "></i>
+              </Button>
+            ) : (
+              <Button variant="outline-dark" onClick={() => handleClick()}>
+                <i className="fas fa-search "></i>
+              </Button>
+            )}
+          </div>
+          <div>
+            <Table responsive>
+              <thead>
+                <tr>
+                  <th>#</th>
+                  <th>First Name</th>
+                  <th>Last Name</th>
+                  <th>Gender</th>
+                  <th>Age</th>
+                  <th>Rank</th>
+                </tr>
+              </thead>
+              <tbody>
+                {student &&
+                  student.map((e, index) => {
+                    if (index < show) {
+                      return (
```

Comment:
Các chỗ return này đang dùng chung ở dưới nhiều, em viết thành 1 hàm để có thể reuse lại.
Mà logic chỗ này if else lồng nhau nhiều quá, có cách nào hơn ko em

Reply:
function Output() {
    return (
      <>
        <tbody>
          {student &&
            student.map((e, index) => {
              return (
                <tr key={index}>
                  <td>{e.id}</td>
                  <td>{firstName(e.full_name)}</td>
                  <td>{lastName(e.full_name)}</td>
                  <td>{gender(e.gender)}</td>
                  <td>{findAge(e)}</td>
                  <td>{e.rank}</td>
                </tr>
              );
            })}
        </tbody>
      </>
    );
  }

Em có viết thêm hàm output như này để reuse nhưng mà không ổn lắm ạ vì mỗi một trường hợp thỏa mãn điều kiện mới return ra kết quả thỏa mãn, nếu mà viết một hàm riêng để reuse thì tới khi gọi lại vào điều kiện nó lại không in ra đúng ạ

Reply:
chỗ if else em có thể xem sau, nhưng chỗ đoạn return
```
return (
                          <tr key={index}>
                            <td>{e.id}</td>
                            <td>{firstName(e.full_name)}</td>
                            <td>{lastName(e.full_name)}</td>
                            <td>{gender(e.gender)}</td>
                            <td>{findAge(e)}</td>
                            <td>{e.rank}</td>
                          </tr>
                        );
```
em có thể tách nó thành 1 hàm như 
 ```
 const renderRowInfo = (data, index) => {
    return (
      <tr key={index}>
        <td>{data.id}</td>
        <td>{firstName(data.full_name)}</td>
        <td>{lastName(data.full_name)}</td>
        <td>{gender(data.gender)}</td>
        <td>{findAge(data)}</td>
        <td>{data.rank}</td>
      </tr>
    );
  };
```
xong ở dưới những chỗ return ra như thế giống nhau chỉ cần 
`return renderRowInfo(e, index);`
trông sẽ gọn gàng và tái sử dụng lại được